systemtest/src/test/java/io/strimzi/systemtest/kafka/listeners/ListenersST.java

```diff
@@ -389,6 +389,90 @@ void testSendMessagesTlsScramSha(ExtensionContext extensionContext) {
+                        .build())
+                .endKafka()
+                .editEntityOperator()
+                .editOrNewTemplate()
+                .editOrNewUserOperatorContainer()
+                .addToEnv(
+                        new ContainerEnvVarBuilder()
+                                .withName("STRIMZI_SCRAM_SHA_PASSWORD_LENGTH")
+                                .withValue(String.valueOf(passwordLength))
+                                .build())
+                .endUserOperatorContainer()
+                .endTemplate()
+                .endEntityOperator()
+                .endSpec()
+                .build());
+
+        resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(clusterName, topicName).build());
+
+        KafkaUser kafkaUser = KafkaUserTemplates.scramShaUser(clusterName, kafkaUsername).build();
+
+        resourceManager.createResource(extensionContext, kafkaUser);
+        resourceManager.createResource(extensionContext, KafkaClientsTemplates.kafkaClients(namespaceName, true, clusterName + "-" + Constants.KAFKA_CLIENTS, kafkaUser).build());
+
+        final String kafkaClientsPodName =
+                kubeClient(namespaceName).listPodsByPrefixInName(namespaceName, clusterName + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+                .withUsingPodName(kafkaClientsPodName)
+                .withTopicName(topicName)
+                .withNamespaceName(namespaceName)
+                .withClusterName(clusterName)
+                .withKafkaUsername(kafkaUsername)
+                .withMessageCount(MESSAGE_COUNT)
+                .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                .build();
+
+        // Check brokers availability
+        LOGGER.info("Checking produced and consumed messages to pod:{}", kafkaClientsPodName);
+
+        internalKafkaClient.checkProducedAndConsumedMessages(
+                internalKafkaClient.sendMessagesTls(),
+                internalKafkaClient.receiveMessagesTls()
+        );
+
+        LOGGER.info("Checking if generated password has {} characters", passwordLength);
+        String password = kubeClient().namespace(namespaceName).getSecret(kafkaUsername).getData().get("password");
+        String decodedPassword = new String(Base64.getDecoder().decode(password));
+
+        assertEquals(decodedPassword.length(), passwordLength);
+
+        Service kafkaService = kubeClient(namespaceName).getService(namespaceName, KafkaResources.bootstrapServiceName(clusterName));
+        String kafkaServiceDiscoveryAnnotation = kafkaService.getMetadata().getAnnotations().get("strimzi.io/discovery");
+        JsonArray serviceDiscoveryArray = new JsonArray(kafkaServiceDiscoveryAnnotation);
+        assertThat(serviceDiscoveryArray, is(StUtils.expectedServiceDiscoveryInfo(9096, "kafka", "scram-sha-512", true)));
```

Comment:
Same as above => this seems to be just a copy paste from another test which is not needed here I guess?